packages/sync-actions/src/product-actions.js

```diff
@@ -234,9 +234,6 @@ export function generateChangeMasterVariantAction (oldObj, newObj) {
   const newMasterVariantId = extractMasterVariantId(newObj)
   const oldMasterVariantId = extractMasterVariantId(oldObj)
 
-  // Previosuly no master variant existed
-  if (!oldMasterVariantId && newMasterVariantId)
```

Comment:
Referring to the comment above and your statement too: this won't happen due to the workings of our API and the implicit generation of a `masterVariant`.